### ISSUE TITLE
Reduce/Remove log warnings for every ignored file

### ISSUE THREAD

**Issue by MilesTrevethan-Datatonic** [sqlfluff/sqlfluff#6003]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

When a file is skipped due to the .sqlfluffignore file a warning is added to the log. This happens for every file that is skipped.
Normally these warnings aren't shown, but when running sqlfluff through pre-commit, if there is an sqlfluff 'FAIL', the entire log output is written out.
When there are a lot of files which were skipped this can spam out the entire output so you can't really see where the the FAIL was that you need to fix.

### Use case

Basically would be good to keep the log output as clean as possible.
Ideas could be to have an option to not show these warnings or just show one warning instead of one warning for each file.

### Dialect

bigquery
dbt-templater

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
Heya @MilesTrevethan-Datatonic 

I've had a good dig into this one - and I don't think there's an easy fix because the warnings you're seeing are there _intentionally_ for CLI users. Because of the way that `pre-commit` calls SQLFluff, we can't tell whether it's being called by a human directly or by `pre-commit`.

I do however have a workaround which I've documented in #6052:

```rst
Ignoring files while using :code:`pre-commit`
---------------------------------------------

Under the hood, `pre-commit`_ works by passing specific files to *SQLFluff*.
For example, if the only two files that are modified in your commit are
:code:`file_a.sql` and :code:`file_b.sql`, then the command which is called
in the background is :code:`sqlfluff lint file_a.sql file_b.sql`. While this
is efficient, it does produce some unwanted noise when also
using :ref:`sqlfluffignore`. This is because *SQLFluff* is designed to allow
users to override an *ignore* configuration by passing the name of the file
directly. This makes a lot of sense in a CLI context, but less so in the context
of being invoked by `pre-commit`_.

To avoid noisy logs when using both `pre-commit`_ and :ref:`sqlfluffignore`,
we recommend also setting the :code:`exclude` argument in your
:code:`.pre-commit-config.yaml` file (either the `top level config`_ or the
`hook specific config`_). This will prevent files matching the given pattern
being passed to *SQLFluff* and so silence any warnings about the
:ref:`sqlfluffignore` being overridden.

.. _`top level config`: https://pre-commit.com/#top_level-exclude
.. _`hook specific config`: https://pre-commit.com/#config-exclude
```

**Comment by MilesTrevethan-Datatonic** [comment]
Hey Alan
Really appreciate you looking into this
That's seems like a good solution, will use it.
Thank you!

**Comment by IngerMathilde** [comment]
Hi, I am now also running into this problem with regards to pre-commit and sqlignore. However, the ignored files are kind of all over, as such using the exclude method mentioned above is quite difficult. Could there be a way to add a flag so that this output is suppressed?

Currently I am using the following hack with a powershell script that calls sqlfluff, but this is really not desirable

```pwsh
# Run the sqlfluff plugin pytests
function Lint-Pre-Commit {
    param(
        [string[]]$Files = @(".")
    )

    Write-Host "🧪 Running linting for SQLFluff"

    # Run sqlfluff with no parallel processes and without the progress bar for cleaner output
    $sqlfluffOutput = sqlfluff lint --processes 0 --disable-progress-bar $Files
    $exitCode = $LASTEXITCODE

    # Filter out specific non-actionable noise:
    # - Suppress lines that start with 'WARNING' or '--disregard-sqlfluffignores',
    #   which relate to files skipped due to ignore patterns. These are expected in pre-commit
    #   workflows and do not indicate real issues. All other output, including other warnings, is retained.
    $filteredOutput = $sqlfluffOutput | Where-Object {
        $_ -notmatch '^(WARNING|--disregard-sqlfluffignores)'
    }
    $filteredOutput

     if ($exitCode -ne 0) {
        Write-Error "❌ SQLFluff found linting issues."
        exit 1
    } else {
        Write-Host "✅ No linting issues found."
    }
}
```

and then a local pre-commit hook

```yaml
# run sqlfluff lint on the changed sql files
- id: sqlfluff-lint
  name: sqlfluff lint
  description: check sql files with sqlfluff
  language: system
  entry: pwsh ./setup.ps1 lint-pre-commit
  types: [ sql ]
  pass_filenames: true
```
